src/main/com/github/perlundq/yajsync/session/Sender.java

```diff
@@ -276,7 +285,8 @@ public Boolean call() throws ChannelException, InterruptedException
             }
             long t3 = System.currentTimeMillis();
 
-            if (_isPreserveUser && _fileSelection != FileSelection.RECURSE) {
+            if (_isPreserveUser && !_isNumericIds
+                    && _fileSelection != FileSelection.RECURSE) {
                 sendUserList();
```

Comment:
Regarding the sent error message from the native rsync receiver (list entry not found) I guess if the sending of the user list should be done although the user data is not required on the receiver side. Do you have another assumption? I didn't check the rsync source code yet.